### PR TITLE
Tpetra: Adding initial memory HWM performance test

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #11743
Tests: This is one
Stakeholder feedback: n/a

### CODE REVIEW

File: packages/tpetra/core/test/ImportExport/ImportExport_GDSWStyle.cpp [hunk 1]
```diff
@@ -0,0 +1,694 @@
...
+  RCP<const Comm<int> > comm = A.getRowMap ()->getComm ();
+  int globalEqual = 0;
+  reduceAll<int, int> (*comm, REDUCE_MIN, 1, &localEqual, &globalEqual);
+  return globalEqual == 1;
+}
+
+
+
+// return current memory usage in kilobytes
+size_t get_memory_usage_now()
```
Comment (cwpearson) [inline packages/tpetra/core/test/ImportExport/ImportExport_GDSWStyle.cpp:356]: Do we care about GPU memory? If so, maybe return a std::pair for system + gpu

**Review by cwpearson — COMMENTED** [review]
I assume the particular problem under test was requested by a user, so I can't speak to that; my only concern is whether this should be augmented to track GPU memory as well.

**Comment by cwpearson** [comment]
Any maybe a squash if you're feeling like it

**Review by cwpearson — APPROVED** [review]

**Comment by ccober6** [comment]
@csiefer2 , yeah it is amazing that the optimized runs on my laptop in <10 seconds, and the debug takes >10 minutes when resources are loaded!

**Pull request merged**